src/data_chain/chain.rs

```diff
@@ -185,3 +171,309 @@ impl Chain {
+        let mut payload = NodeState {
+            state: State::ElderLive,
+            public_key: *node_a.public_info().sign_key(),
+            age: node_a.public_info().age(),
+            section: prefix,
+        };
+        let live_a_block = tests::create_block(payload.clone(), &nodes, false);
+        let last_node = &nodes[nodes.len() - 1];
+        assert!(!live_a_block.get_node_infos().contains(
+            last_node.public_info(),
+        ));
+        let live_a_vote = unwrap!(Vote::new(nodes[0].secret_sign_key(), payload));
+
+        // Create Live(B) and extra vote for Live(B).
+        let node_b = FullInfo::within_range(6, &prefix.lower_bound(), &prefix.upper_bound());
+        payload = NodeState {
+            state: State::ElderLive,
+            public_key: *node_b.public_info().sign_key(),
+            age: node_b.public_info().age(),
+            section: prefix,
+        };
+        let live_b_block = tests::create_block(payload.clone(), &nodes, false);
+        assert!(!live_b_block.get_node_infos().contains(
+            last_node.public_info(),
+        ));
+        let live_b_vote = unwrap!(Vote::new(nodes[0].secret_sign_key(), payload));
+
+        // Create chain containing Live(A) in `blocks` and Live(B) in `pending_blocks`.
+        let mut chain = Chain::from_blocks(
+            vec![live_a_block.clone()],
+            vec![live_b_block.clone()],
+            nodes.len(),
+        );
+
+        // Add the vote for Live(A) and check it gets added to the Live(A) block.
+        let mut proof = Proof {
+            node_info: *last_node.public_info(),
+            sig: *live_a_vote.signature(),
+        };
+        chain.add_vote(live_a_vote, *last_node.public_info());
+        assert_eq!(chain.blocks.len(), 1);
+        assert!(chain.blocks[0].proofs().contains(&proof));
+        assert_eq!(chain.pending_blocks.len(), 1);
+        assert!(!chain.pending_blocks[0].proofs().contains(&proof));
+
+        // Add the vote for Live(B) and check it gets added to the Live(B) block.
+        proof.sig = *live_b_vote.signature();
+        chain.add_vote(live_b_vote, *last_node.public_info());
+        assert_eq!(chain.blocks.len(), 1);
+        assert!(!chain.blocks[0].proofs().contains(&proof));
+        assert_eq!(chain.pending_blocks.len(), 1);
+        assert!(chain.pending_blocks[0].proofs().contains(&proof));
+    }
+
+    #[ignore]
+    #[test]
+    fn equality() {
+        // Create one chain comprising:
+        //     Live(A)→Live(B)→Offline(A)→Live(C)→Relocated(B)→Live(D)
+        // and another comprising:
+        //     Live(A)→Live(B)→Relocated(B)→Live(C)→Offline(A)→Live(D)
+        let mut rng = SeededRng::thread_rng();
+        unwrap!(rust_sodium::init_with_rng(&mut rng));
+
+        let prefix = Prefix::default().pushed(rng.gen()).with_version(rng.gen());
+        let nodes = tests::create_full_infos(Some(prefix));
```

Comment:
just to confirm this `nodes` won't contain any node of A, B, C, D, which means those node won't be signer for any of the blocks.

Reply:
Yes, that's right, A, B, C, and D don't sign anything.  I thought the test was simpler that way.  I toyed with making the blocks more complete by having A-D signing those blocks which "made sense" for them to sign, but I think it'd make the test way more complex with little benefit from the point of view of testing chain equality.